gateway/ln-gateway/src/lib.rs

```diff
@@ -1545,6 +1560,234 @@ pub(crate) async fn fetch_lightning_node_info(
     Ok((node_pub_key, alias, network))
 }
 
+impl Gateway {
+    async fn public_key(&self, federation_id: &FederationId) -> Option<PublicKey> {
```

Comment:
Can we rename this to `gateway_client_public_key`? Or `gateway_redeem_key`? Right now the gateway also has a public key for the lightning node and a public key for the `gateway_id`. Lots of public keys to keep track of!